pytest/lib/mocknet_helpers.py

```diff
@@ -9,34 +9,40 @@
 RPC_PORT = '3030'
 
 
-def get_status():
-    r = requests.get(f'http://{LOCAL_ADDR}:{RPC_PORT}/status', timeout=10)
+def get_status(addr=LOCAL_ADDR, port=RPC_PORT):
+    r = requests.get(f'http://{addr}:{port}/status', timeout=10)
     r.raise_for_status()
     return r.json()
 
 
-def json_rpc(method, params):
+def json_rpc(method, params, addr=LOCAL_ADDR, port=RPC_PORT):
     j = {'method': method, 'params': params, 'id': 'dontcare', 'jsonrpc': '2.0'}
-    r = requests.post(f'http://{LOCAL_ADDR}:{RPC_PORT}', json=j, timeout=10)
+    r = requests.post(f'http://{addr}:{port}', json=j, timeout=10)
     return r.json()
 
 
-def get_nonce_for_pk(account_id, pk, finality='optimistic'):
-    access_keys = json_rpc(
-        'query', {
-            'request_type': 'view_access_key_list',
-            'account_id': account_id,
-            'finality': finality
-        })
+def get_nonce_for_pk(account_id,
+                     pk,
+                     finality='optimistic',
+                     addr=LOCAL_ADDR,
+                     port=RPC_PORT):
+    access_keys = json_rpc('query', {
+        'request_type': 'view_access_key_list',
+        'account_id': account_id,
+        'finality': finality
+    },
+                           addr=addr,
+                           port=port)
     logger.info(f'get_nonce_for_pk {account_id}')
     assert access_keys['result']['keys'], account_id
     for k in access_keys['result']['keys']:
         if k['public_key'] == pk:
             return k['access_key']['nonce']
```

Comment:
Not related to this PR, but shouldn’t this be something like:

    nonce = next((key['access_key']['nonce']
                  for key in access_keys['result']['keys']
                  if key['public_key'] == pk), None)
    if nonce is None:
        raise KeyError(f'Nonce for {account_id} {pk} not found')
    return nonce

Reply:
I meant that the previous loop is now unnecessary with this, i.e.:

```python
def get_nonce_for_pk(account_id,
                     pk,
                     finality='optimistic',
                     addr=LOCAL_ADDR,
                     port=RPC_PORT):
    request = {
        'request_type': 'view_access_key_list',
        'account_id': account_id,
        'finality': finality
    }
    access_keys = json_rpc('query', request, addr=addr, port=port)
    logger.info(f'get_nonce_for_pk {account_id}')
    nonce = next((key['access_key']['nonce']
                  for key in access_keys['result']['keys']
                  if key['public_key'] == pk), None)
    if nonce is None:
        raise KeyError(f'Nonce for {account_id} {pk} not found')
    return nonce
```

Or perhaps I don’t understand the point of the `assert`?  Current code
checks if account has keys and then looks for one matching the
argument.  If the account has no keys than it raises AssertionError
but if it has keys but no key matches given one it returns None.

It felt to me like the intention was to raise an exception when the
nounce cannot be found?  Which is what my proposed code does.

Reply:
Thanks